packages/typespec-ts/test/modularUnit/operations.spec.ts

```diff
@@ -121,8 +121,12 @@ describe("operations", () => {
           ...Bar): OkResponse;
           `;
 
-      const operationFiles =
-        await emitModularOperationsFromTypeSpec(tspContent);
+      const operationFiles = await emitModularOperationsFromTypeSpec(
+        tspContent,
+        {
+          mustEmptyDiagnostic: false
```

Comment:
what diagnostic warning will we get? is it the query format deprecation warning? but why it's not reported before query format should be deprecated a long time ago ?

Reply:
this is a bugfix and here we should report diagnostic of `unable-serialized-type` for header but not: https://github.com/marygao/autorest.typescript/blob/uri-template-support/packages/typespec-ts/src/utils/parameterUtils.ts#L152-L156.

so the new warning is introduced by this pr's fix.